website/docs/general/contributor-guide.md

```diff
@@ -58,55 +54,83 @@ Good First Issue curates easy pickings from this project, and helps you make you
 - [Apache APISIX® Java Plugin Runner](https://github.com/apache/apisix-java-plugin-runner/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
 - [Apache APISIX® Go Plugin Runner](https://github.com/apache/apisix-go-plugin-runner/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
 
-### Create your branch
+## Git Flow
+
+To contribute code or documentation, setup your local machine for development.
 
-Switch to your forked master branch, pull codes from upstream, then create a new branch.
+### Fork the repo and clone your fork
+
+Fork the [Apache APISIX repo](https://github.com/apache/apisix/) and clone your fork to your local machine.
 
 ```sh
-$ git checkout master
-$ git pull upstream master
-$ git checkout -b IssueNo
+git clone https://github.com/your-username/apisix.git
 ```
 
-Notice: We will merge PR using squash, commit logs will be different form upstream if you use one older branch.
-
-### Coding
+### Add 'upstream' repo to list of remotes
 
-1. Please obey the Code of Conduct during the process of development and finish the check before submitting the pull request.
+```sh
+git remote add upstream https://github.com/apache/apisix.git
+```
 
-2. Then push codes to your fork repo.
+You can then verify the configured remotes.
 
 ```sh
-$ git add modified-file-names
-$ git commit -m 'commit message'
-$ git push origin issueNo
+git remote -v
 ```
 
-### Submit PR
+Learn more about Git and GitHub flow by following the [GitHub Docs](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
 
-1. Send a pull request to the master branch.
+**Note**: Setup your username and email address to make sure that your ID shows up in the contributor list.
 
-2. The mentor will do codes review before discussing some details (including the design, the implementation and the performance) with you.
+```sh
+git config --global user.name "username"
+git config --global user.email "mail address"
+```
 
-3. Also make sure that the pull request title has a semantic prefix like `fix:` or `feat:` or any other [conventional commit types](https://github.com/commitizen/conventional-commit-types/blob/master/index.json).
+### Create your branch
 
-4. Then congratulate to you to be an official contributor of Apache APISIX.
+To keep your local fork up to date, fetch and merge the upstream remote.
```

Comment:
Direct merge can introduce extra merge commits in your pull request if it cannot be fast-forwarded. A better and preferred way is to always rebase the pull request branch against the `upstream/main`. If not rebased successfully it will show conflicts. Solve the conflicts and then continue with rebase. This method also ensures less work for the developer merging the PR.

Reply:
I agree. My usual flow is that before I create a feature branch, I fetch the upstream/master and merge it with my local master and push it to origin. Once my local master and upstream master is same, I create a feature branch. Then, while working on the feature branch, I rebase it with upstream/master to make sure that there aren't any conflicts and I am working on top of the latest changes.

Reply:
Yeah, that is nice too, we just have to make sure not to introduce unnecessary commits, whatever serves that is fine.